src/rust/engine/src/nodes.rs

```diff
@@ -513,13 +520,16 @@ impl Node for Select {
       );
 
     let variant_value = variant_value.map(|s| s.to_string());
-    let node = self.clone();
     deps_future
       .and_then(move |dep_results| {
-        future::result(node.choose_task_result(context, dep_results, &variant_value))
+        future::result(self.choose_task_result(context, dep_results, &variant_value))
       })
       .boxed()
   }
+
+  fn is_inline() -> bool {
+    return true
```

Comment:
The `return` statement is only necessary when you want to return early. In all of these cases, just `true`/`false` is fine.